app/templates/frameworks/agreement.html

```diff
@@ -52,96 +52,130 @@
+                    }
+                  ]
+                },
+                {
+                  "lead_in": "If you can’t digitally sign your framework agreement:",
+                  "sublists": [
+                    {
+                      "text": "Print and sign page 5."
+                    },
+                    {
+                      "text": "Scan the page and save as PDF, JPG or PNG."
+                    },
+                    {
+                      "text": "Send your signed framework agreement to CCS by uploading it below."
+                    }
+                  ]
+                },
+              ]
+            }
+          ],
+          verbose = true
+        %}
+          {% include "toolkit/instruction-list.html" %}
+        {% endwith %}
 
-                {%
+        <form method="post" enctype="multipart/form-data" action="{{ url_for(".upload_framework_agreement", framework_slug=framework.slug) }}">
+            <input type="hidden" name="csrf_token" value="{{ csrf_token() }}" />
+            {% if supplier_framework.agreementReturned %}
+              {%
                 with
-                type = "save",
-                label = "Send document to CCS"
-                %}
-                {% include "toolkit/button.html" %}
-                {% endwith %}
-            </form>
-        </div>
+                start = 4,
+                verbose = 'True',
+                items = [
+                  {
+                    "body": "Upload your signed framework agreement",
+                    "uploads": [
+                      {
+                          "value": "Document uploaded {}".format(supplier_framework.agreementReturnedAt),
+                          "name": "agreement",
+                          "file_type": "PDF",
+                          "error": upload_error
+                      }
+                    ]
+                  }
+                ]
+              %}
+                {% include "toolkit/instruction-list.html" %}
+              {% endwith %}
+            {% else %}
+              {%
+                with
+                start = 4,
+                verbose = 'True',
+                items = [
+                  {
+                    "body": "Upload your signed framework agreement",
+                    "uploads": [
+                      {
+                          "value": "",
+                          "name": "agreement",
+                          "file_type": "PDF",
+                          "error": upload_error
+                      }
+                    ]
+                  }
+                ]
+              %}
+                {% include "toolkit/instruction-list.html" %}
+              {% endwith %}
+            {% endif %}
+
+            {%
+            with
+            type = "save",
+            label = "Send document to CCS"
+            %}
+            {% include "toolkit/button.html" %}
```

Comment:
This button "belongs" to the final item in the instruction list really (the choose-a-file element), so it seems odd to have the button completely removed from that.  I think this is where the extra space is coming from too.